src/bindgen/language_backend/java_jna.rs

```diff
@@ -0,0 +1,860 @@
+        match l {
+            Literal::Expr(expr) => {
+                write!(out, "{expr}")
+            }
+            Literal::Struct { export_name, .. } => {
+                // There is an hashmap in there that doesn't have stable debug output
+                not_implemented(&format!("Struct Literal {export_name}"), out)
+            }
+            _ => not_implemented(l, out),
+        }
+    }
+}
+
+enum JnaIntegerType {
+    Byte,
+    Short,
+    Int,
+    NativeLong,
+    Long,
+    SizeT,
+}
+
+impl JnaIntegerType {
+    pub fn size(&self) -> &str {
+        match self {
+            JnaIntegerType::Byte => "1",
+            JnaIntegerType::Short => "2",
+            JnaIntegerType::Int => "4",
+            JnaIntegerType::NativeLong => "Native.LONG_SIZE",
+            JnaIntegerType::Long => "8",
+            JnaIntegerType::SizeT => "Native.SIZE_T_SIZE",
+        }
+    }
+
+    pub fn set_method(&self) -> &str {
+        match self {
+            JnaIntegerType::Byte => "setByte(0, (byte)value.intValue())",
+            JnaIntegerType::Short => "setShort(0, (short)value.intValue())",
+            JnaIntegerType::Int => "setInt(0, value.intValue())",
+            JnaIntegerType::NativeLong | JnaIntegerType::SizeT => {
+                "setNativeLong(0, new NativeLong(value.longValue()))"
+            }
+            JnaIntegerType::Long => "setLong(0, value.longValue())",
+        }
+    }
+
+    pub fn get_method(&self) -> &str {
+        match self {
+            JnaIntegerType::Byte => "getByte(0)",
+            JnaIntegerType::Short => "getShort(0)",
+            JnaIntegerType::Int => "getInt(0)",
+            JnaIntegerType::NativeLong | JnaIntegerType::SizeT => "getNativeLong(0).longValue()",
+            JnaIntegerType::Long => "getLong(0)",
+        }
+    }
+
+    pub fn from_kind(kind: &IntKind) -> Self {
+        match kind {
+            IntKind::Short => JnaIntegerType::Short,
+            IntKind::Int => JnaIntegerType::Int,
+            IntKind::Long => JnaIntegerType::NativeLong,
+            IntKind::LongLong => JnaIntegerType::Long,
+            IntKind::SizeT => JnaIntegerType::SizeT,
+            IntKind::Size => JnaIntegerType::SizeT,
+            IntKind::B8 => JnaIntegerType::Byte,
+            IntKind::B16 => JnaIntegerType::Short,
+            IntKind::B32 => JnaIntegerType::Int,
+            IntKind::B64 => JnaIntegerType::Long,
+        }
+    }
+}
+
+struct JnaStruct<'a> {
+    documentation: &'a Documentation,
+    constants: &'a Vec<(&'a Constant, &'a Struct)>,
+    fields: &'a Vec<Field>,
+    name: &'a str,
+    superclass: &'a str,
+    interface: &'a str,
+    deprecated: &'a Option<String>,
+}
+
+struct IndexedFunctionArg<'a> {
+    ty: &'a Type,
+    name: &'a Option<String>,
+    index: usize,
+}
+
+impl JavaJnaLanguageBackend<'_> {
+    fn write_deprecated<F: Write>(
+        &mut self,
+        out: &mut SourceWriter<F>,
+        deprecated: &Option<String>,
```

Comment:
Let's use `Option<&str>` (and deprecated.as_deref() in the callers)

Reply:
:+1: